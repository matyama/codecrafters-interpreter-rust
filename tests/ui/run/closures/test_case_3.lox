fun makeLogger(prefix) {
  var logCount = 0;

  fun log(message) {
    logCount = logCount + 1;
    print prefix + ": " + message;

    if (logCount > 6) {
      print prefix + ": Too many log lines!";
      logCount = 0;
    }
  }

  return log;
}

var debugLog = makeLogger("debug");
var errorLog = makeLogger("error");

debugLog("Starting");
debugLog("Processing");
debugLog("Finishing");
debugLog("Extra line");

errorLog("Failed!");
errorLog("Retrying...");
