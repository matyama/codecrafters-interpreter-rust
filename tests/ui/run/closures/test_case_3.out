debug: Starting
debug: Processing
debug: Finishing
debug: Extra line
error: Failed!
error: Retrying...
